### PR TITLE
WT-6475 Adding a statistic to track how many times we call cursor pre…

### PR DESCRIPTION
**Pull request by luke-pearson**
…v when we see a globally visible tombstone

### CODE REVIEW

File: src/txn/txn_rollback_to_stable.c [hunk 1]
```diff
@@ -1087,8 +1090,11 @@ __rollback_to_stable_btree_hs_truncate(WT_SESSION_IMPL *session, uint32_t btree_
          * we can skip it.
          */
         if (__wt_txn_visible_all(
-              session, cbt->upd_value->tw.stop_txn, cbt->upd_value->tw.durable_stop_ts))
+              session, cbt->upd_value->tw.stop_txn, cbt->upd_value->tw.durable_stop_ts)) {
+            WT_STAT_CONN_INCR(session, cursor_prev_hs_tombstone);
+            WT_STAT_DATA_INCR(session, cursor_prev_hs_tombstone);
```
Comment (tetsuo-cpp) [inline src/txn/txn_rollback_to_stable.c:1095]: Are you sure that this is going to be incrementing for the data store's dhandle (as opposed to the history store table)?

Comment (luke-pearson) [reply src/txn/txn_rollback_to_stable.c]: Good point, occasionally we do run these operations with the history store dhandle as opposed to the data store one as such i'll remove it from that.

File: src/txn/txn_rollback_to_stable.c [hunk 2]
```diff
@@ -232,8 +232,10 @@ __rollback_row_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW
          * we can skip it.
          */
         if (__wt_txn_visible_all(
-              session, cbt->upd_value->tw.stop_txn, cbt->upd_value->tw.durable_stop_ts))
+              session, cbt->upd_value->tw.stop_txn, cbt->upd_value->tw.durable_stop_ts)) {
+            WT_STAT_CONN_INCR(session, cursor_prev_hs_tombstone);
```
Comment (agorrod) [inline src/txn/txn_rollback_to_stable.c:236]: I think it'd be useful to split the rollback to stable counters into a different statistic.

File: dist/stat_data.py [hunk 3]
```diff
@@ -365,6 +365,7 @@ def __init__(self, name, desc, flags=''):
     CursorStat('cursor_next_skip_lt_100', 'cursor next calls that skip less than 100 entries'),
     CursorStat('cursor_next_skip_total', 'Total number of entries skipped by cursor next calls'),
     CursorStat('cursor_prev', 'cursor prev calls'),
+    CursorStat('cursor_prev_hs_tombstone', 'cursor prev calls that were the result of a history store tombstone'),
```
Comment (agorrod) [inline dist/stat_data.py:368]: Can you make this: "cursor prev calls that skip due to a globally visible history store tombstone"?

File: src/txn/txn_rollback_to_stable.c (hunk 2 above)
Comment (luke-pearson) [reply src/txn/txn_rollback_to_stable.c:236]: Yeah I thought about that but it raised the question of how many statistics we'd need to differentiate well, hopefully two will be enough.

**Pull request merged**